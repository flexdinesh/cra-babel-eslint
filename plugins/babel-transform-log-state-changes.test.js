import { transformSync } from "@babel/core";
import logStateChangesPlugin from "./babel-transform-log-state-changes";

const HelloWorld = `
const HelloWorld = () => {
  const [val, setVal] = useState(true);
  const [anotherVal, setAnotherVal] = useState(100);
  return <span>Hello World</span>;
};
`;

test("inject state change wrapper", () => {
  const { code } = transformSync(HelloWorld, {
    ast: true,
    plugins: ["@babel/plugin-transform-react-jsx", logStateChangesPlugin]
  });

  expect(code).toMatchSnapshot();
});
