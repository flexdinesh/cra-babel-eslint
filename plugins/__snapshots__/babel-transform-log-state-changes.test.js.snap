// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inject state change wrapper 1`] = `
"const customSetVal = (setter, cName, sName) => val => {
  if (!global.stateStore) {
    global.stateStore = {};
  }

  const str = global.stateStore;

  if (str[cName]) {
    if (str[cName][sName]) {
      str[cName][sName].push(val);
    } else {
      str[cName][sName] = [val];
    }
  } else {
    str[cName] = {
      [sName]: [val]
    };
  }

  const vals = str[cName][sName];

  if (vals.length > 1) {
    const prevVal = vals[vals.length - 2];
    console.log(\\"%c >> StateChangeLog >>\\", \\"color: green;\\", cName, \\":\\", sName, \\":\\", prevVal, \\"=>\\", val);
  } else {
    console.log(\\"%c >> StateChangeLog >>\\", \\"color: green;\\", cName, \\":\\", sName, \\":\\", val);
  }

  return setter(val);
};

const HelloWorld = () => {
  let [val, setVal] = useState(true);
  const setVal_INTERNAL = customSetVal(setVal, \\"HelloWorld\\", \\"val\\");
  setVal = setVal_INTERNAL
  let [anotherVal, setAnotherVal] = useState(100);
  const setAnotherVal_INTERNAL = customSetVal(setAnotherVal, \\"HelloWorld\\", \\"anotherVal\\");
  setAnotherVal = setAnotherVal_INTERNAL
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;
